### PR TITLE
run the async mq send message using a thread safe asyncio API. & add task cleanup.

### PR DESCRIPTION
**Pull request by deadly-panda**

### CODE REVIEW

File: src/ostorlab/agent/agent.py [hunk 1]
```diff
@@ -187,7 +193,14 @@ def run(self) -> None:
             if self.in_selectors is not None and len(self.in_selectors) > 0:
                 logger.debug("starting mq run")
                 self._loop.run_until_complete(self.mq_run())
+                for sig in (signal.SIGINT, signal.SIGTERM):
+                    self._loop.add_signal_handler(sig, _at_exit_loop)
                 self._loop.run_forever()
+                tasks = asyncio.all_tasks()
+                for task in [t for t in tasks if not (t.done() or t.cancelled())]:
```
Comment (mohsinenar) [inline src/ostorlab/agent/agent.py:200]:
```suggestion
                for task in [t for t in tasks if (t.done() or t.cancelled()) is False]:
```

**Review by rabsondev — APPROVED** [review]

File: src/ostorlab/agent/agent.py [hunk 2]
```diff
@@ -67,6 +67,12 @@ def _setup_logging(agent_key: str, universe: str) -> None:
             )
 
 
+def _at_exit_loop() -> None:
+    """Cancel all tasks when receiving a SIGINT (Interrupt from keyboard) or a SIGTERM (termination)."""
+    for task in asyncio.all_tasks():
+        task.cancel()
```
Comment (3asm) [inline src/ostorlab/agent/agent.py:73]:
```suggestion
def _cancel_tasks() -> None:
    """Cancel all tasks when receiving a SIGINT (Interrupt from keyboard) or a SIGTERM (termination)."""
    for task in asyncio.all_tasks():
        task.cancel()
```

File: src/ostorlab/agent/agent.py [hunk 3]
```diff
@@ -187,7 +193,14 @@ def run(self) -> None:
             if self.in_selectors is not None and len(self.in_selectors) > 0:
                 logger.debug("starting mq run")
                 self._loop.run_until_complete(self.mq_run())
+                for sig in (signal.SIGINT, signal.SIGTERM):
+                    self._loop.add_signal_handler(sig, _at_exit_loop)
```
Comment (3asm) [inline src/ostorlab/agent/agent.py:197]:
```suggestion
                self._loop.add_signal_handler(signal.SIGTERM, _at_exit_loop)
                self._loop.add_signal_handler(signal.SIGINT, _at_exit_loop)
```

File: src/ostorlab/agent/agent.py [hunk 4]
```diff
@@ -187,7 +193,14 @@ def run(self) -> None:
...
                 self._loop.run_forever()
+                tasks = asyncio.all_tasks()
+                for task in [t for t in tasks if (t.done() or t.cancelled()) is False]:
+                    # give canceled tasks the last chance to run.
+                    self._loop.run_until_complete(task)
```
Comment (3asm) [inline src/ostorlab/agent/agent.py:202]: List comprehension if a for loop. At this point just use an if.

**Review by 3asm — CHANGES_REQUESTED** [review]

**Review by 3asm — CHANGES_REQUESTED** [review]

**Review by 3asm — APPROVED** [review]

**Pull request merged**